doc.go

```diff
@@ -0,0 +1,95 @@
+// Generated by running
+//  go generate github.com/gonum/matrix
+// DO NOT EDIT.
+
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package matrix provides common error handling mechanisms for matrix operations
+// in mat64 and cmat128.
+//
+// Overview
+//
+// This section provides a quick overview of the matrix package. The following
+// sections provide more in depth commentary.
+//
+// matrix provides:
+//  - Error type definitions
+//  - Error recovery mechanisms
+//  - Common constants used by mat64 and cmat128
+//
+// Errors
+//
+// The mat64 and cmat128 matrix packages share a common set of errors
+// provided by matrix via the matrix.Error type.
+//
+// Errors are either returned directly or used as the parameter of a panic
+// depending on the class of error encountered.
```

Comment:
Maybe it'll help to say when we panic.

"Errors are either returned directly or used as the parameter of a panic depending on the class of error encountered. Examples of each type are found in the mat64 Solve methods, which find x such that A *x = b.  

Panics:
A function will panic when the input parameters are inappropriate for the function or when correctness is easily verified before calling the function. In Solve, for example, the columns of each matrix must be equal. Solve will panic with ErrShape if this is violated, as column length is easily verified before the call to solve

Error Returns:
An error value is returned from the function or method when the operation can meaningfully fail. The Solve operation is cannot complete if A is singular. However, determining the singularity of A is most easily discovered during the Solve procedure itself, and so in this case an error is returned."

Not great, but maybe something along those lines?